ui/components/ui/hardware-wallet-state/hardware-wallet-state.js

```diff
@@ -0,0 +1,65 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import {
+  getHardwareWalletDevice,
+  getHardwareWalletPath,
+} from '../../../selectors';
+import { isDeviceAccessible } from '../../../store/actions';
+import { I18nContext } from '../../../contexts/i18n';
+import { HARDWARE_CHECK_RATE } from '../../../../shared/constants/hardware-wallets';
+import ActionableMessage from '../actionable-message';
+
+const noOp = () => {
+  // do nothing
+};
+
+/**
+ * Component that monitors hardware wallet state and displays a warning if locked
+ *
+ * @param options0
+ * @param options0.pollingRateMs
+ * @param options0.initialStatus
+ * @param options0.onUpdate
+ */
+export default function HardwareWalletState({
+  pollingRateMs = HARDWARE_CHECK_RATE,
+  initialStatus = 'locked',
+  onUpdate = noOp,
+}) {
+  const t = useContext(I18nContext);
+  const [status, setStatus] = useState(initialStatus);
+  const device = useSelector(getHardwareWalletDevice);
+  const path = useSelector(getHardwareWalletPath);
+
+  const updateHardwareLockState = useCallback(async () => {
+    const unlocked = await isDeviceAccessible(device, path);
+    const state = unlocked ? 'unlocked' : 'locked';
+    setStatus(state);
+    onUpdate(state);
+  }, [device, path, onUpdate]);
+
+  useEffect(() => {
+    const intervalId = setInterval(updateHardwareLockState, pollingRateMs);
+    return () => clearInterval(intervalId);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [pollingRateMs]);
+
+  return (
+    status === 'locked' && (
+      <ActionableMessage
+        message={t('ledgerLocked')}
+        type="danger"
+        className="hardware-wallet-state__error"
+        useIcon
+        iconFillColor="var(--color-error-default)"
+      />
```

Comment:
You should be able to use BannerAlert here. ActionableMessage is in the process of being deprecated. I'll create an issue to deprecate ActionableMessage.

https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-banneralert--severity

Reply:
Thanks @georgewrmarshall - I'll make that change later today

Reply:
Thanks @flexa-rob will try re-review today